src/test/java/org/opensearch/security/support/Base64CustomHelperTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.support;
+
+import com.amazon.dlic.auth.ldap.LdapUser;
+import org.junit.Assert;
+import org.junit.Test;
+import org.ldaptive.LdapEntry;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.security.auth.UserInjector;
+import org.opensearch.security.user.AuthCredentials;
+import org.opensearch.security.user.User;
+
+import java.io.Serializable;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static org.opensearch.security.support.Base64CustomHelper.deserializeObject;
+import static org.opensearch.security.support.Base64CustomHelper.serializeObject;
+
+public class Base64CustomHelperTest {
+
+    private static Serializable ds(Serializable s) {
+        return deserializeObject(serializeObject(s));
+    }
+
+    @Test
+    public void testString() {
+        String string = "string";
+        Assert.assertEquals(string, ds(string));
+    }
+
+    @Test
+    public void testInteger() {
+        Integer integer = Integer.valueOf(0);
+        Assert.assertEquals(integer, ds(integer));
+    }
+
+    @Test
+    public void testDouble() {
+        Double number = Double.valueOf(0.);
+        Assert.assertEquals(number, ds(number));
+    }
+
+    @Test
+    public void testInetSocketAddress() {
+        InetSocketAddress inetSocketAddress = new InetSocketAddress(0);
+        Assert.assertEquals(inetSocketAddress, ds(inetSocketAddress));
+    }
+
+    @Test
+    public void testUser() {
+        User user = new User("user");
+        Assert.assertEquals(user, ds(user));
+    }
+
+    @Test
+    public void testSourceFieldsContext() {
+        SourceFieldsContext sourceFieldsContext = new SourceFieldsContext(new SearchRequest(""));
+        Assert.assertEquals(sourceFieldsContext.toString(), ds(sourceFieldsContext).toString());
+    }
+
+    @Test
+    public void testHashMap() {
+        HashMap map = new HashMap();
```

Comment:
can we add a case with non empty HashMap?

Reply:
Done